src/hooks/tradingView/useChartLines.ts

```diff
@@ -0,0 +1,200 @@
+import { useEffect, useState } from 'react';
+
+import { shallowEqual, useSelector } from 'react-redux';
+
+import { AbacusOrderStatus } from '@/constants/abacus';
+import { STRING_KEYS } from '@/constants/localization';
+import { type OrderType, ORDER_TYPE_STRINGS } from '@/constants/trade';
+import type { ChartLine, ChartLineType, TvWidget } from '@/constants/tvchart';
+
+import { useStringGetter } from '@/hooks';
+
+import { getCurrentMarketOrders, getCurrentMarketPositionData } from '@/state/accountSelectors';
+import { getAppTheme, getAppColorMode } from '@/state/configsSelectors';
+
+import { MustBigNumber } from '@/lib/numbers';
+import { getChartLineColors } from '@/lib/tradingView/utils';
+
+let chartLines: Record<string, ChartLine> = {};
+
+/**
+ * @description Hook to handle drawing chart lines
+ */
+
+export const useChartLines = ({
+  tvWidget,
+  displayButton,
+  isChartReady,
+}: {
+  tvWidget: TvWidget | null;
+  displayButton: HTMLElement | null;
+  isChartReady?: boolean;
+}) => {
+  const [showOrderLines, setShowOrderLines] = useState(false);
+
+  const stringGetter = useStringGetter();
+
+  const appTheme = useSelector(getAppTheme);
+  const appColorMode = useSelector(getAppColorMode);
+
+  const currentMarketPositionData = useSelector(getCurrentMarketPositionData, shallowEqual);
+  const currentMarketOrders = useSelector(getCurrentMarketOrders, shallowEqual);
+
+  useEffect(() => {
+    if (isChartReady && displayButton) {
+      displayButton.onclick = () => {
+        const newShowOrderLinesState = !showOrderLines;
+        if (newShowOrderLinesState) {
+          displayButton?.classList?.add('order-lines-active');
+        } else {
+          displayButton?.classList?.remove('order-lines-active');
+        }
+        setShowOrderLines(newShowOrderLinesState);
+      };
+    }
+  }, [isChartReady, showOrderLines]);
+
+  useEffect(() => {
+    if (tvWidget && isChartReady) {
+      tvWidget.onChartReady(() => {
+        tvWidget.chart().dataReady(() => {
+          if (showOrderLines) {
+            drawOrderLines();
+            drawPositionLine();
+          } else {
+            deleteChartLines();
+          }
+        });
+      });
+    }
+  }, [isChartReady, showOrderLines, currentMarketPositionData, currentMarketOrders]);
+
+  const drawPositionLine = () => {
+    if (!currentMarketPositionData) return;
+
+    const { current: entryPrice } = currentMarketPositionData.entryPrice;
+    const { current: quantity } = currentMarketPositionData.size;
```

Comment:
<img width="604" alt="Screen Shot 2024-02-13 at 10 18 51 AM" src="https://github.com/dydxprotocol/v4-web/assets/13111994/1ebbfeb6-69a2-4313-b4fd-8b10b75cc1ab">

TypeError

Reply:
oh huh I didn't realize typescript/javascript validation wasn't on in my code editor by default 😢 sry sry will fix